gammapy/visualization/utils.py

```diff
@@ -178,3 +181,135 @@ def plot_theta_squared_table(table):
+    kwargs_axes : dict
+        Keyword arguments to pass to `matplotlib.axes.Axes`.
+    kwargs_fit : dict
+        Keyword arguments to pass to `scipy.optimize.curve_fit`
+
+    Returns
+    -------
+    result_list : list of dict
+        List of the results of `scipy.optimize.curve_fit` as a  dictionary, where `param` is the best-fit
+        parameters of `func`, `covar` is the covariance matrix for these parameters, and `info_dict` is the return of
+        the same name of `scipy.optimize.curve_fit`.
+    axes : `~numpy.ndarray` of `~matplotlib.pyplot.Axes`
+        Array of Axes.
+
+    Examples
+    --------
+    >>> from gammapy.datasets import MapDataset
+    >>> from gammapy.estimators import TSMapEstimator
+    >>> from scipy.stats import norm
+    >>> from gammapy.visualization import plot_distribution
+    >>> dataset = MapDataset.read("$GAMMAPY_DATA/cta-1dc-gc/cta-1dc-gc.fits.gz")
+    >>> tsmap_est = TSMapEstimator().run(dataset)
+    >>> func = lambda x, mu, sig : norm.pdf(x, loc=mu, scale=sig)
+    >>> res, ax = plot_distribution(tsmap_est.sqrt_ts, func=func, kwargs_hist={'bins': 75, 'range': (-10, 10), 'density': True})
+    """
+
+    from gammapy.maps import WcsNDMap  # import here to avoid circular import
+
+    if not isinstance(wcs_map, WcsNDMap):
+        raise TypeError(
+            f"map_ must be an instance of gammapy.maps.WcsNDMap, given {type(wcs_map)}"
+        )
+
+    kwargs_hist = kwargs_hist or {}
+    kwargs_axes = kwargs_axes or {}
+    kwargs_fit = kwargs_fit or {}
+
+    kwargs_hist.setdefault("density", True)
+    kwargs_fit.setdefault("full_output", True)
+
+    if fit and func is None:
+        raise ValueError(
+            "Must provide a function to fit with parameter fit set to True."
+        )
+
+    cutout, mask = wcs_map.cutout_and_mask_region()
+    idx_x, idx_y = np.where(mask)
+
+    data = cutout.data[..., idx_x, idx_y]
+
+    if ax is None:
+        n_plot = len(data)
+        cols = min(ncols, n_plot)
+        rows = 1 + (n_plot - 1) // cols
+
+        width = 12
+        figsize = (width, width * rows / cols)
+
+        fig, axes = plt.subplots(
+            nrows=rows,
+            ncols=cols,
+            figsize=figsize,
+        )
+        cells_in_grid = rows * cols
+    else:
+        axes = ax
+        cells_in_grid = len(ax.flatten())
+
+    if not isinstance(axes, np.ndarray):
+        axes = np.array([axes])
+
+    result_list = []
+
+    for idx in range(cells_in_grid):
+
+        axe = axes.flat[idx]
+        if idx > len(data) - 1:
+            axe.set_visible(False)
+            continue
+        d = data[idx][np.isfinite(data[idx])]
+        n, bins, _ = axe.hist(d, **kwargs_hist)
+
+        if fit:
+            centers = 0.5 * (bins[1:] + bins[:-1])
+
+            pars, cov, infodict, message, _ = curve_fit(func, centers, n, **kwargs_fit)
+
+            result_dict = {"param": pars, "covar": cov, "info_dict": infodict}
```

Comment:
to be useful in case of cubes, the dict might need the axes indices or their edges/center

Reply:
Done ! Not sure of the naming I put though.